benches/benches/block_target_gas_set/contract.rs

```diff
@@ -1,20 +1,741 @@
+    //   of gas at 100_000.
+    // let size = 2620_u32; // 18bit integer maxes at 262144
+    // let contract: Vec<_> = (0..100_u32)
+    //     .map(|x| x * size)
+    //     .map(|x| vec![op::movi(0x13, x), op::scwq(0x13, 0x29, 0x14)]) // copy range starting at $rA of size $rC
+    //     .flatten()
+    //     .collect();
+    // let gas = 100_000;
+    // let instructions = vec![
+    //     op::gtf_args(0x10, 0x00, GTFArgs::ScriptData),
+    //     op::addi(0x11, 0x10, ContractId::LEN.try_into().unwrap()),
+    //     op::addi(0x11, 0x11, WORD_SIZE.try_into().unwrap()),
+    //     op::addi(0x11, 0x11, WORD_SIZE.try_into().unwrap()),
+    //     op::movi(0x12, gas),
+    //     op::movi(0x14, size),
+    //     op::call(0x10, RegId::ZERO, 0x11, 0x12),
+    // ];
+    // replace_contract_in_service(&mut service, &contract_id, contract);
+    // run_with_service(
+    //     "contract/scwq",
+    //     group,
+    //     instructions,
+    //     script_data.clone(),
+    //     &service,
+    //     contract_id,
+    //     &rt,
+    //     &mut rng,
+    // );
+
+    // srw
+
+    {
+        let contract = vec![op::srw(0x13, 0x14, 0x15), op::ret(RegId::ZERO)];
+        let mut instructions = setup_instructions();
+        instructions.extend(vec![
+            op::movi(0x15, 2000),
+            op::call(0x10, RegId::ZERO, 0x11, 0x12),
+            op::jmpb(RegId::ZERO, 0),
+        ]);
+        replace_contract_in_service(&mut service, &contract_id, contract);
+        run_with_service(
+            "contract/srw",
+            group,
+            instructions,
+            script_data.clone(),
+            &service,
+            contract_id,
+            &rt,
+            &mut rng,
+        );
+    }
+
+    // srwq
+
+    // TODO: This is under-costed, so it runs too long and will complete before running out of gas
+    // let size = 2620_u32;
+    // let contract = (0..2620)
+    //     .map(|x| x * size)
+    //     .map(|x| vec![op::movi(0x13, x), op::srwq(0x14, 0x29, 0x13, 0x15)])
+    //     .flatten()
+    //     .collect();
+    // let gas = 100_000;
+    // let instructions = vec![
+    //     op::gtf_args(0x10, 0x00, GTFArgs::ScriptData),
+    //     op::addi(0x11, 0x10, ContractId::LEN.try_into().unwrap()),
+    //     op::addi(0x11, 0x11, WORD_SIZE.try_into().unwrap()),
+    //     op::addi(0x11, 0x11, WORD_SIZE.try_into().unwrap()),
+    //     op::movi(0x12, gas),
+    //     op::movi(0x15, size),
+    //     op::call(0x10, RegId::ZERO, 0x11, 0x12),
+    // ];
+    // replace_contract_in_service(&mut service, &contract_id, contract);
+    // run_with_service(
+    //     "contract/srwq",
+    //     group,
+    //     instructions,
+    //     script_data.clone(),
+    //     &service,
+    //     contract_id,
+    //     &rt,
+    //     &mut rng,
+    // );
+
+    // sww
+
+    {
+        let contract = vec![op::sww(RegId::ZERO, 0x29, RegId::ONE), op::ret(RegId::ZERO)];
```

Comment:
Why do you want to return here? backward jump looks good=)

Reply:
Also, it would be nice to write different values into the storage to generate different hash. You can increment some value by 1 each iteration